api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,93 @@
+package org.pmiops.workbench.api;
+
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.leonardo.model.LeonardoDiskStatus;
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final LeonardoMapper leonardoMapper;
+  private final Provider<DbUser> userProvider;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceDao workspaceDao,
+      Provider<DbUser> userProvider,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceDao = workspaceDao;
+    this.userProvider = userProvider;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    String googleProject = lookupWorkspace(workspaceNamespace).getGoogleProject();
+    try {
+      LeonardoListPersistentDiskResponse response;
+      String pdNamePrefix = userProvider.get().getUserPDNamePrefix();
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false).stream()
+              .filter(r -> r.getName().startsWith(pdNamePrefix))
+              .sorted((r1, r2) -> r2.getId().compareTo(r1.getId())) // sort by descent order
+              .collect(Collectors.toList());
+
+      if (responseList.isEmpty()) {
+        return ResponseEntity.notFound().build();
+      }
+      if (responseList.size() > 1) {
+        log.warning(String.format("Observed multiple active PDs for a single user"));
+      }
+      response = responseList.get(0);
+      if (LeonardoDiskStatus.FAILED.equals(response.getStatus())) {
+        log.warning(
+            String.format(
+                "Observed Leonardo persistent disk with unexpected failed status:\n%s",
```

Comment:
workspace namespace is also helpful information in this log. Otherwise, developer has not idea which workspace goes wrong just by given status